src/Microsoft.Diagnostics.EventFlow.Inputs.MicrosoftLogging/Logger/EventFlowLogger.cs

```diff
@@ -102,27 +106,39 @@ private LogLevel ToLogLevel(Extensions.Logging.LogLevel loggerLevel)
             return LogLevel.Verbose;
         }
 
-        private void ApplyToAllButLast(object src, System.Action<KeyValuePair<string, object>> action)
+        private bool ApplyOmittingUnformattedMessage(object src, System.Action<KeyValuePair<string, object>> action)
         {
+            // Special case the "{OriginalFormat}" data key that is used to preserve the message before formatting was applied
+            // This data is omitted from the payload and replaced by formatted message.
+            // See https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs
+            const string OriginalFormatKey = "{OriginalFormat}";
+
+            // If the data is not structured, it will be omitted
+            bool unformattedMessageOmitted = true;
+
             var structuredSrc = src as IEnumerable<KeyValuePair<string, object>>;
             if (structuredSrc != null)
             {
+                unformattedMessageOmitted = false;
                 using (var srcEnumerator = structuredSrc.GetEnumerator())
```

Comment:
Why not just use `foreach` on `IEnumerable<T1,T2>`?

Reply:
That was refactored from original code that needed to use the enumerator directly. Fixed now!